lib/srv/server/ec2_watcher.go

```diff
@@ -349,8 +318,8 @@ func newEC2InstanceFetcher(cfg ec2FetcherConfig) *ec2InstanceFetcher {
 		Values: []string{string(ec2types.InstanceStateNameRunning)},
 	}}
 
-	if _, ok := cfg.Labels["*"]; !ok {
-		for key, val := range cfg.Labels {
+	if _, ok := cfg.Matcher.Tags["*"]; !ok {
```

Comment:
Suggestion: Prefer using the constant where possible
```suggestion
	if _, ok := cfg.Matcher.Tags[types.Wildcard]; !ok {
```